### ISSUE TITLE
Expense - Error when deleting expense when Delay submissions is enabled

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49157]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.34-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N/A - no delete option in Admin side
**Issue was found when executing this PR:** https://github.com/Expensify/App/pull/48998
**Email or phone of affected tester (no customers):** applausetester+kh010901@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Admin and employee are in the same workspace.
- Delay submissions is enabled in Worflows page.

1. Go to staging.new.expensify.com
2. [Employee] Submit an expense in the workspace chat, but do not click the Submit button.
3. [Admin] Go to transaction thread in the workspace chat.
4. [Admin] Click on the report header.
5. [Admin] Click Delete expense.
6. [Admin] Delete the expense.

## Expected Result:
The expense should be deleted because delete option is present.
On prod, delete option is not available.

## Actual Result:
The expense is not deleted and error shows up.
Also, when selecting the same expense in Search, there is no delete option from the dropdown but it is present in the transaction thread.

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/3d375b61-51bc-4ea8-880e-567901035421

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by flodnv** [comment]
Hmmm, this solution implies that the owner also won't be able to delete the transaction, which I think is wrong?

**Comment by luacmartins** [comment]
[Logs](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%22DeleteMoneyRequest%22%20AND%20email%3A%22applausetester%2Bkh11090001%40applause.expensifail.com%22%20AND%20timestamp%3A%5B2024-09-12T00%3A00%20TO%202024-09-14T23%3A59%5D) FYI the email provided in the OP seems incorrect, the correct one should be applausetester+kh11090001@applause.expensifail.com

```
Throw ExpException - 8478b963e0cf0e66800a56c0d4ea36bc ~~ message: '401 Unauthorized' exceptionMessage: 'Auth DeleteMoneyRequest returned an error' exceptionFile: '/git/releases/expensify.com/a07a885/lib/Auth.php' exceptionLine: '128' exceptionCode: '401'
```

So this means we prevent admins from deleted unsubmitted expenses, which I think is correct. So we should update the logic to hide the button for the admin as we have in prod.

**Comment by Pujan92** [comment]
@daledah I think we need to add an extra condition of report is in the open state or not with `isPolicyAdmin`.

https://github.com/Expensify/App/blob/478611cdeedf3a90af3a4dfb160e7e57a98a3e57/src/pages/ReportDetailsPage.tsx#L202

```js
(isPolicyAdmin && !ReportUtils.isOpenExpenseReport(moneyRequestReport))
```

**Comment by trjExpensify** [comment]
Following the blame, pulling this one out to #wave-control for a second: https://expensify.slack.com/archives/C06ML6X0W9L/p1726235265572009

**Comment by trjExpensify** [comment]
So I'm not sure where this came from, wires must have been crossed, but to confirm on the expected behaviour... 

Admins should not be able to delete other people's expenses in NewDot. They can delete their own, but for others, they can put them on hold, not delete them. So please can we revert whatever made that possible in all places; 

- Search page
- Request preview component action menu 
- RHP details of an expense 
- Anywhere else I've forgotten, but I think that's it! 😅 

Thanks!

**Comment by luacmartins** [comment]
Probably this PR? https://github.com/Expensify/App/pull/48998

**Comment by luacmartins** [comment]
I think we should revert https://github.com/Expensify/App/pull/48998 and then work on a backend fix to prevent admins from being able to delete someone else's expenses.

**Comment by luacmartins** [comment]
Revert here https://github.com/Expensify/App/pull/49178

**Comment by Pujan92** [comment]
> Probably this PR? #48998

Yes @luacmartins, bcoz as per the [confirmation](https://github.com/Expensify/App/issues/46762#issuecomment-2339040992) by @dylanexpensify we allowed admin to delete the expenses. But as it seems to be wrong behavior we need to revert the PR as you suggested.

**Comment by luacmartins** [comment]
I looked at the backend logic and I'm not sure why the search page would return `true` in that case

**Comment by luacmartins** [comment]
Regardless, once the CP is done this is no longer an App blocker since the backend logic hasn't changed in a while

**Comment by flodnv** [comment]
Hmmm IIRC, admins can delete (remove) expenses from reports in order to not approve them -- of course, that was before we implemented "hold".

**Comment by trjExpensify** [comment]
Yeah, we don't have that concept in NewDot.

**Comment by luacmartins** [comment]
This is fixed on staging. The Search part is still an issue, so I'll keep the original issue open since we still need to address that.
